src/torch_linear_assignment_cuda_kernel.cu

```diff
@@ -265,6 +271,7 @@ std::vector<torch::Tensor> batch_linear_assignment_cuda(torch::Tensor cost) {
 
   AT_DISPATCH_FLOATING_TYPES(cost.scalar_type(), "solve_cuda_batch", [&] {
     solve_cuda_batch<scalar_t>(
+        cost.scalar_type(),
```

Comment:
note: if we initialize `u`, `v`, `shortestPathCosts` before invoking `solve_cuda_batch`, then we don't have to pass scalar_type as argument. I took this approach because the diff is easier to see.